src/components/cardModal/CardModal.js

```diff
@@ -0,0 +1,141 @@
+import styles from "@/styles/cardModal.module.scss";
+import { useState, useContext } from "react";
+import { LoggedInContext } from "@/pages/_app";
+
+
+async function sendCardInfo(userName, cardInfo)
+{
+    const resp = await fetch('/api/users/addcard', {
+        "method": "PATCH",
+        "headers": {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        },
+        "body": JSON.stringify({
+            "userName": userName,
+            "paymentMethods": cardInfo
+        })
+    })
+    const data = await resp.json();
+
+    return data;
+}
+
+
+
+export default function CardModal2({ showCardModal, handleCloseCardModal, handleCard })
+{
+    const [cardInfo, setCardInfo] = useState({ "cardNr": '', "validTo": '', "CVC": '', "nameOnCard": '' })
+    const {
+        isLoggedIn,
+        setIsLoggedIn,
+        username,
+        setUsername,
+        setDisplayPopup,
+        setPopupTitle,
+        setPopupMessage
+    } = useContext(LoggedInContext);
+
+    if (!showCardModal)
+        return null;
+
+    return (
+        <>
+            <div className={styles.cardModalOverlay} onClick={handleCloseCardModal} />
+
+            <div className={styles.cardModalContainer}>
+                <button
+                    className={styles.closeBtn}
+                    onClick={handleCloseCardModal}
+                    aria-label="Stäng modalen knapp"
+                >X</button>
+
+                <form className={styles.cardModalForm} onSubmit={(ev) =>
+                {
+                    ev.preventDefault();
+                    handleCard(cardInfo);
+
+                    if (isLoggedIn)
+                    {
+                        const resp = sendCardInfo(username, cardInfo);
```

Comment:
This will return immediately, before the server has had a chance to process the request and decide whether it works or not, because `sendCardInfo()` is an async function, which means that it returns a `Promise`, not the data that the promise may eventually resolve to.

I think you want an `await` here, which will give you the resulting data instead of the promise.